pkg/pool-utils/contracts/BasePool.sol

```diff
@@ -226,8 +226,16 @@ abstract contract BasePool is
         _miscData = _miscData.insertBool(enabled, _RECOVERY_MODE_BIT_OFFSET);
 
         emit RecoveryModeStateChanged(enabled);
+
+        if (!enabled) _onDisableRecoveryMode();
```

Comment:
Add a comment that we're calling a hook? We already have a warning up in RecoveryMode about not putting any "dangerous" code that might fail here, especially on the enable path. Maybe worth repeating, since we are actually doing something at this level, and more derived contracts might start here?